### PR TITLE
WIP: Master merge

### PR DESCRIPTION
**Pull request by richardjgowers**
This merges the 1.x branch into develop again.

### CODE REVIEW

File: testsuite/MDAnalysisTests/analysis/test_waterdynamics.py [hunk 1]
```diff
@@ -40,6 +40,18 @@ def universe():
...
+        universe, SELECTION1, SELECTION1, 0, 5, 3)
+    hbl.run()
+    assert_almost_equal(hbl.timeseries[2][1], 0.75, 5)
+
+
+def test_HydrogenBondLifetimes_deprecation(universe):
+    with pytest.deprecated_call():
+        hbl = waterdynamics.HydrogenBondLifetimes(
+            universe, SELECTION1, SELECTION1, 0, 5, 3)
+
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/analysis/test_waterdynamics.py:54]:
```suggestion
```
Removed in #2842

File: testsuite/MDAnalysisTests/analysis/test_wbridge.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from io import StringIO
 from collections import defaultdict
+from six.moves import reload_module
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/analysis/test_wbridge.py:3]:
```suggestion
```

File: testsuite/MDAnalysisTests/analysis/test_wbridge.py [hunk 3]
```diff
@@ -765,3 +766,8 @@ def test_warn_results_deprecated(self, universe_DA):
         wmsg = "The `timeseries` attribute was deprecated in MDAnalysis 2.0.0"
         with pytest.warns(DeprecationWarning, match=wmsg):
             assert_equal(wb.timeseries, wb.results.timeseries)
+
+def test_import_warning():
+    wmsg = 'This module has been moved to'
+    with pytest.warns(DeprecationWarning, match=wmsg):
+        reload_module(MDAnalysis.analysis.hbonds.wbridge_analysis)
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/analysis/test_wbridge.py:773]:
```suggestion
```
stub has now been removed

File: testsuite/MDAnalysisTests/analysis/test_wbridge.py [hunk 4]
```diff
@@ -751,7 +752,7 @@ def test_duplicate_water(self):
         wb = WaterBridgeAnalysis(u, 'resname LEU and name O',
                                  'resname LEU and name N H', order=4)
         wb.run()
-        assert len(wb.results.timeseries[0]) == 2
+        assert len(wb.timeseries[0]) == 2
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/analysis/test_wbridge.py:755]:
```suggestion
        assert len(wb.results.timeseries[0]) == 2
```

File: testsuite/MDAnalysisTests/converters/test_parmed.py [hunk 5]
```diff
@@ -295,6 +295,13 @@ def test_equivalent_coordinates(self, ref, output):
         assert_almost_equal(ref.coordinates, output.coordinates, decimal=3)
 
 
+def test_pass_ts_error():
+    u = mda.Universe(PDB_small)
+    err = "Writing Timesteps to ParmEd objects is not supported"
+    with pytest.raises(ValueError, match=err):
+        c = ParmEdConverter()
+        c.convert(u.trajectory.ts)
+
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/converters/test_parmed.py:304]:
```suggestion
```
@lilyminium please double check

File: testsuite/MDAnalysisTests/coordinates/test_netcdf.py [hunk 6]
```diff
@@ -918,3 +918,11 @@ def test_wrong_n_atoms(self, outfile):
             u = make_Universe(trajectory=True)
             with pytest.raises(IOError):
                 w.write(u)
+
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/coordinates/test_netcdf.py:921]: We're good to keep this for now, I'll fix this later in #3294

File: testsuite/MDAnalysisTests/coordinates/test_pdb.py [hunk 7]
```diff
@@ -1281,3 +1281,12 @@ def test_cryst_meaningless_select():
...
     assert cur_sele.n_atoms == 0
+
+
+def test_deprecated_chainid_from_segid(tmpdir):
+    u = mda.Universe(PDB)
+    err = ("Using the last letter of the segid for the chainID "
+           "is now deprecated and will be changed in 2.0. ")
+    with tmpdir.as_cwd():
+        with pytest.warns(DeprecationWarning, match=err):
+            u.atoms.write("test.pdb")
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/coordinates/test_pdb.py:1292]: This should be fixed now right? @lilyminium @jbarnoud 
```suggestion
```

File: testsuite/MDAnalysisTests/core/test_atomgroup.py [hunk 8]
```diff
@@ -1042,6 +1042,84 @@ def ag(self, request):
         return group
 
     @pytest.fixture()
+    def unordered_ag(self, ag):
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/core/test_atomgroup.py:1045]: @mnmelo can you check this please? As far as I'm aware all these were changed in #3012?

File: testsuite/MDAnalysisTests/core/test_universe.py [hunk 9]
```diff
@@ -1291,3 +1292,20 @@ def test_empty_creation_raises_error(self):
...
+def test_deprecate_b_tempfactors():
+    u = mda.Universe(PDB)
+    with pytest.warns(DeprecationWarning, match="alias"):
+        u.add_TopologyAttr("bfactors")
+
+
+def test_deprecate_temp_bfactors():
+    u = mda.Universe(MMTF)
+    with pytest.warns(DeprecationWarning, match="alias"):
+        u.add_TopologyAttr("tempfactors")
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/core/test_universe.py:1311]: @lilyminium is this still necessary?

File: testsuite/MDAnalysisTests/lib/test_distances.py [hunk 10]
```diff
@@ -84,8 +84,8 @@ def test_capped_distance_noresults():
 
 npoints_1 = (1, 100)
 
-boxes_1 = (np.array([1, 2, 3, 90, 90, 90], dtype=np.float32),  # ortho
-           np.array([1, 2, 3, 30, 45, 60], dtype=np.float32),  # tri_box
+boxes_1 = (np.array([10, 20, 30, 90, 90, 90], dtype=np.float32),  # ortho
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/lib/test_distances.py:87]: these are all changes introduced with the new nsgrid code right?

File: testsuite/MDAnalysisTests/topology/test_guessers.py [hunk 11]
```diff
@@ -131,9 +131,19 @@ def test_guess_impropers():
     assert_equal(len(vals), 12)
 
 
+def bond_sort(arr):
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/topology/test_guessers.py:134]: @richardjgowers  Why was this needed again? nsgrid stuff?

**Review by IAlibay — CHANGES_REQUESTED** [review]
Here are all my comments, where I didn't know I've pinged folks who are most likely to know.

Can a couple of the @MDAnalysis/coredevs please double check? This is is our top priority as merging this is key to releasing the beta.

File: maintainer/conda/MDAnalysis/meta.yaml [hunk 12]
```diff
@@ -1,12 +1,12 @@
...
   # This has to be changed after a release
-  version: "0.19.1dev"
+  version: "1.1.1"
 
 source:
    git_url: https://github.com/MDAnalysis/mdanalysis
-   git_branch: develop
-   # git_tag: release-0.15.0
+   # git_branch: develop
+   git_tag: release-0.19.1
```
[the remark this answers is not in the document]
Comment (richardjgowers) [reply maintainer/conda/MDAnalysis/meta.yaml:9]: I think it used to be how we did conda releases pre conda-forge, so could probably be deleted tbh

File: package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py [hunk 13]
```diff
@@ -768,7 +768,8 @@ def count_by_time(self):
         indices /= self.step
 
         counts = np.zeros_like(self.frames)
-        counts[indices.astype(int)] = tmp_counts
+        counts[indices.astype(np.intp)] = tmp_counts
```
[the remark this answers is not in the document]
Comment (richardjgowers) [reply package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py:771]: Yeah I think this is the 32/64 bit compliant way of saying the integer that addresses anything (i.e. `.ix` is `intp`).  I think `int` had previously led to 32/64 mismatches

File: maintainer/conda/MDAnalysis/meta.yaml (hunk 12 above)
[the remark this answers is not in the document]
Comment (IAlibay) [reply maintainer/conda/MDAnalysis/meta.yaml:9]: let's just leave it as it is, we can always worry about this later on 🙃

File: package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (IAlibay) [reply package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py:771]: So the difference here was just between what Tyler and I wrote for the numpy 1.20 deprecation stuff, as long as it works and isn't significantly difference in terms of performance let's just go with it.

File: package/MDAnalysis/coordinates/PDB.py [hunk 14]
```diff
@@ -903,6 +903,10 @@ def write(self, obj):
             The :class:`~MDAnalysis.core.groups.AtomGroup` or
             :class:`~MDAnalysis.core.universe.Universe` to write.
         """
+        warnings.warn("Using the last letter of the segid for the chainID "
+                      "is now deprecated and will be changed in 2.0. "
+                      "In 2.0, the chainID attribute will be used if it "
+                      "exists, or a placeholder value.", DeprecationWarning)
```
[the remark this answers is not in the document]
Comment (IAlibay) [reply package/MDAnalysis/coordinates/PDB.py]: Found it, adding a docstring entry to clarify the current situation.

File: testsuite/MDAnalysisTests/topology/test_guessers.py (hunk 11 above)
Comment (richardjgowers) [reply testsuite/MDAnalysisTests/topology/test_guessers.py:134]: Yeah so `self_capped_distance` used to always sort bond indices as a side effect of the algorithm, it no longer does.  So guessed bonds are now in an arbitrary order.

File: testsuite/MDAnalysisTests/converters/test_parmed.py (hunk 5 above)
Comment (IAlibay) [reply testsuite/MDAnalysisTests/converters/test_parmed.py:304]: Yeah ok, we decided to remove that in #3240

File: package/MDAnalysis/core/selection.py [hunk 15]
```diff
@@ -600,6 +612,46 @@ def apply(self, group):
...
+    field = 'moltypes'
+
+
+class SegmentNameSelection(_ProtoStringSelection):
+    """Select atoms based on 'segids' attribute"""
+    token = 'segid'
+    field = 'segids'
+    level = 'segindices'
+
+
```
[the remark this answers is not in the document]
Comment (lilyminium) [reply package/MDAnalysis/core/selection.py:642]: Yes, it should be replaced by gen_selection_class :)

File: package/MDAnalysis/core/universe.py [hunk 16]
```diff
@@ -99,6 +99,8 @@
 from .topologyattrs import AtomAttr, ResidueAttr, SegmentAttr
 from .topologyobjects import TopologyObject
 
+from ..lib.util import deprecate
```
[the remark this answers is not in the document]
Comment (lilyminium) [reply package/MDAnalysis/core/universe.py:102]: I think this `decorate` was for `as_Universe`, which I think is gone in develop now? so I don't think we need the deprecate decorator

File: testsuite/MDAnalysisTests/core/test_universe.py (hunk 9 above)
Comment (lilyminium) [reply testsuite/MDAnalysisTests/core/test_universe.py:1311]: for alias bfactors == tempfactors? I guess I'd say leave the warning in until we actually alias them

File: testsuite/MDAnalysisTests/core/test_atomgroup.py (hunk 8 above)
Comment (mnmelo) [reply testsuite/MDAnalysisTests/core/test_atomgroup.py:1045]: Do you mean the non-seeded RNG use? I agree it should be stabilized.

File: package/MDAnalysis/core/groups.py [hunk 17]
```diff
@@ -988,7 +988,8 @@ def center(self, weights, pbc=False, compound='group', unwrap=False):
 
         # Unwrap Atoms
         if unwrap:
-            coords = atoms.unwrap(compound=comp, reference=None, inplace=False)
+            coords = atoms.unwrap(compound=comp, reference=None,
```
[the remark this answers is not in the document]
Comment (mnmelo) [reply package/MDAnalysis/core/groups.py:991]: Indeed! There's not even a `sort_indices` in this scope anymore.

File: testsuite/MDAnalysisTests/core/test_atomgroup.py (hunk 8 above)
Comment (IAlibay) [reply testsuite/MDAnalysisTests/core/test_atomgroup.py:1045]: No, I meant more, out of all this code from `master` does any of this need to be kept or was it all intentionally removed in `develop`?

File: package/MDAnalysis/analysis/gnm.py [hunk 18]
```diff
@@ -231,6 +231,17 @@ class GNMAnalysis(AnalysisBase):
...
+
+    Attributes
+    ----------
+    results : list
+          eigenvalues and eigenvectors
+
+          .. deprecated:: 1.1.0
+             The structure of the ``results`` list will change in
+             MDAnalysis 2.0.
+
```
[the remark this answers is not in the document]
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/gnm.py:244]: Yes, you can remove this. The docs of `gnm` are rewritten in the develop branch.

File: package/MDAnalysis/analysis/lineardensity.py [hunk 19]
```diff
@@ -67,6 +68,10 @@ class LinearDensity(AnalysisBase):
     results.x.slice_volume : float
            volume of bin in [xyz] direction
 
+          .. deprecated:: 1.1.0
+             The structure of the ``results`` dictionary will change in
+             MDAnalysis 2.0
+
```
[the remark this answers is not in the document]
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/lineardensity.py:74]: Yes!

File: testsuite/MDAnalysisTests/analysis/test_gnm.py [hunk 20]
```diff
@@ -134,3 +134,11 @@ def test_closeContactGNMAnalysis_weights_None(universe):
...
        0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
        0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, -43.0, -3.0])
+
+
+def test_results_warning(universe):
+  msg = ("The structure of the `results` list will change in MDAnalysis "
+          "version 2.0.")
+  with pytest.warns(DeprecationWarning, match=msg):
+        mda.analysis.gnm.GNMAnalysis(universe).run()
+
```
[the remark this answers is not in the document]
Comment (PicoCentauri) [reply testsuite/MDAnalysisTests/analysis/test_gnm.py:144]: Yes!

**Review by PicoCentauri — CHANGES_REQUESTED** [review]
Thanks for merging everything back. You can safely remove the warnings.

File: package/MDAnalysis/analysis/gnm.py (hunk 18 above)
[the remark this answers is not in the document]
Comment (IAlibay) [reply package/MDAnalysis/analysis/gnm.py:244]: cheers

**Review by IAlibay — APPROVED** [review]
Assuming CI comes back green, lgtm.

I can't speak for the nsgrid sections so I'm assuming you're happy with all the changes @richardjgowers

File: testsuite/MDAnalysisTests/core/test_universe.py [hunk 21]
```diff
@@ -1291,3 +1292,20 @@ def test_empty_creation_raises_error(self):
         with pytest.raises(TypeError) as exc:
             u = mda.Universe()
         assert 'Universe.empty' in str(exc.value)
+
+
+def test_as_Universe_deprecation():
+    with pytest.deprecated_call():
+        _ = mda.core.universe.as_Universe(PSF, DCD)
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/core/test_universe.py:1299]:
```suggestion
```
Needs removing

File: package/setup.py [hunk 22]
```diff
@@ -432,7 +439,7 @@ def extensions(config):
                               extra_compile_args=encore_compile_args)
     nsgrid = MDAExtension('MDAnalysis.lib.nsgrid',
                              ['MDAnalysis/lib/nsgrid' + cpp_source_suffix],
-                             include_dirs=include_dirs,
+                             include_dirs=include_dirs + ['MDAnalysis/lib/include'],
```
[the remark this answers is not in the document]
Comment (richardjgowers) [reply package/setup.py:435]: Yeah this is because we now reuse a component of calc_distances in nsgrid

File: package/MDAnalysis/lib/include/calc_distances.h [hunk 23]
```diff
@@ -43,7 +43,28 @@ static void minimum_image(double* x, float* box, float* inverse_box)
   }
 }
 
-static void minimum_image_triclinic(double* dx, float* box)
+inline void _minimum_image_ortho_lazy(double* x, float* box, float* half_box)
```
[the remark this answers is not in the document]
Comment (richardjgowers) [reply package/MDAnalysis/lib/include/calc_distances.h:46]: Yeah this is just an optimised version for nsgrid, for when you know the coordinates are at most a single box length away

File: package/MDAnalysis/lib/distances.py [hunk 24]
```diff
@@ -1096,9 +1122,9 @@ def _nsgrid_capped_self(reference, max_cutoff, min_cutoff=None, box=None,
             gridsearch = FastNS(max_cutoff, reference, box=box)
             results = gridsearch.self_search()
 
-        pairs = results.get_pairs()[::2, :]
+        pairs = results.get_pairs()
```
[the remark this answers is not in the document]
Comment (richardjgowers) [reply package/MDAnalysis/lib/distances.py:1127]: Yeah everything used to be double added, then deduplicated

**Comment by richardjgowers** [comment]
Yeah I did a sweep, LGTM

**Review by PicoCentauri — COMMENTED** [review]
LGTM
